File: app/src/main/java/com/monodev/PAMS/MyNotificationListener.java
package com.monodev.PAMS;
import android.app.Notification;
import android.os.Build;
import android.os.Bundle;
import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;

import com.google.firebase.messaging.RemoteMessage;


public class MyNotificationListener extends NotificationListenerService {
    public final static String TAG = "PAMS_listener";

    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        super.onNotificationPosted(sbn);
        Notification notification = sbn.getNotification();
        Bundle extras = sbn.getNotification().extras;
        String message = sbn.getGroupKey();
        boolean b1 = message.contains("com.monodev.pams");

        if(b1 == false) {
            CharSequence text = extras.getCharSequence(Notification.EXTRA_TEXT);
            POST myTask = new POST(this);
            myTask.execute((String) text);
        }
    }
}

File: app/src/main/java/com/monodev/PAMS/MyNotificationListener.kt
package com.monodev.PAMS

import android.app.Notification
import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification

class MyNotificationListener : NotificationListenerService() {
    override fun onNotificationPosted(sbn: StatusBarNotification) {
        super.onNotificationPosted(sbn)
        val notification = sbn.notification
        val extras = sbn.notification.extras
        val message = sbn.groupKey
        val b1 = message.contains("com.monodev.pams")
        val text = extras.getCharSequence(Notification.EXTRA_TEXT)
        if (b1 == false) {
            if(text is String){
                val myTask = POST(this)
                val _text = text.replace("[\n\r]".toRegex(), "")
                myTask.execute(_text as String?)
            }
        }
    }
}